core/src/main/java/com/graphhopper/routing/profiles/MappedDecimalEncodedValue.java

```diff
@@ -63,18 +65,53 @@ private int toInt(double val) {
         return (int) Math.round(val / precision);
     }
 
+    int findClosestIndex(int val) {
+        int res = Arrays.binarySearch(toValueMapArray, val);
+        if (res >= 0)
+            throw new IllegalStateException("Cannot happen as toStorageMap.getOrDefault should have caught this case");
+
+        // TODO NOW rounding up from e.g. max_height=10 to 11 might be ugly. Not sure how or what to do about it or if we should make it even configurable.
```

Comment:
Maybe adding a precise mode would make sense. Every value that cannot be exactly matched will throw an error or return a fail flag in `setDecimal`. 

I actually assumed this class would work like that. If you map Strings, you would probably fail and not search for the closest String :).

Reply:
I'm unsure if we need two modes. Actually the naming "MappedDecimal" might be wrong as it is more a compressed "fequency based" storage format, but not losless.

Do you know a use case where you would need the precise mode?

Reply:
Yes probably renaming MappedDecimal would make this clearer.

Right now I don't have a good example where the precise mode would be necessary. One could always extend the list of stored values, if 1 or 2 values should be available. 

I think my issues was that I expected it to work like a mapped String.

Reply:
I just had an idea where a precise mode could make sense. Maybe this could allow us to map `Double.INFINITY` and `Double.NaN`, with NaN referring to the default value and INFINITY referring to unlimited speed :).

Reply:
Yeah, exactly that problem appears when we want to introduce a default value (INFINITY). But my problem is that in reality you want it both: 
 * if the stored value is 0 (the default) then return INFINITY. 
 * For everything else find the closest entry. 

Having a simple MappedDecimal class would be much simpler than it is now, but then the "compactness" advantages over the FactoredDecimalEncodedValue would disappear IMO, especially for large areas with many distinct value.

Reply:
> Double.INFINITY and Double.NaN

I would use INFINITY as default value, i.e. map the 0-index to infinity (all vehicle allowed by default)

Another idea could be to collect the values "on-the-go" while import and throw an exception if the bits (array size) are too small. And prefill this array with one value: infinity.

But maybe the best solution for now would be to remove this class and use a FactoredDecimalEncodedValue? The current "mapped"-approach uses 5 bits. A "factor"-approach with factor=0.1 and max=25 means 8 bits. Do you think this is acceptable for now?